pkg/skaffold/sync/kubectl/kubectl.go

```diff
@@ -60,18 +58,30 @@ func (k *Syncer) Sync(ctx context.Context, s *sync.Item) error {
 	return nil
 }
 
-func deleteFileFn(ctx context.Context, pod v1.Pod, container v1.Container, src, dst string) []*exec.Cmd {
-	delete := exec.CommandContext(ctx, "kubectl", "exec", pod.Name, "--namespace", pod.Namespace, "-c", container.Name, "--", "rm", "-rf", dst)
+func deleteFileFn(ctx context.Context, pod v1.Pod, container v1.Container, files map[string]string) []*exec.Cmd {
+	// "kubectl" is below...
+	deleteCmd := []string{"exec", pod.Name, "--namespace", pod.Namespace, "-c", container.Name, "--", "rm", "-rf"}
+	args := make([]string, 0, len(deleteCmd)+len(files))
+	args = append(args, deleteCmd...)
+	for _, dst := range files {
+		args = append(args, dst)
+	}
+	delete := exec.CommandContext(ctx, "kubectl", args...)
 	return []*exec.Cmd{delete}
 }
 
-func copyFileFn(ctx context.Context, pod v1.Pod, container v1.Container, src, dst string) []*exec.Cmd {
-	dir := filepath.Dir(dst)
-	var cmds []*exec.Cmd
-	if _, ok := syncedDirs[dir]; !ok {
-		cmds = []*exec.Cmd{exec.CommandContext(ctx, "kubectl", "exec", pod.Name, "-c", container.Name, "-n", pod.Namespace, "--", "mkdir", "-p", dir)}
-		syncedDirs[dir] = struct{}{}
-	}
-	copy := exec.CommandContext(ctx, "kubectl", "cp", src, fmt.Sprintf("%s/%s:%s", pod.Namespace, pod.Name, dst), "-c", container.Name)
-	return append(cmds, copy)
+func copyFileFn(ctx context.Context, pod v1.Pod, container v1.Container, files map[string]string) []*exec.Cmd {
+	// Use "m" flag to touch the files as they are copied.
+	reader, writer := io.Pipe()
+	copy := exec.CommandContext(ctx, "kubectl", "exec", pod.Name, "--namespace", pod.Namespace, "-c", container.Name, "-i",
```

Comment:
Remove the tar after extraction?

Also TIL about `-m` or `--touch` flag. Cool

Reply:
The tar file is piped from Go's memory directly to `kubectl exec`'s stdin.  So there's nothing to remove.

Reply:
Ah missed that 👍